src/transaction/log_replication_atomic.cpp

```diff
@@ -367,4 +370,26 @@ namespace cublog
 	  }
       }
   }
+
+  void
+  atomic_replicator::release_locks_by_tranid (cubthread::entry &thread_entry, const TRANID trid)
+  {
+    assert (m_locked_objects.count (trid) > 0);
+
+    for (auto it = m_locked_objects.lower_bound (trid); it != m_locked_objects.upper_bound (trid); it++)
```

Comment:
Nit. It would be more verbose but seems clearer to me.
```suggestion
    auto [begin, end] = m_locked_objects.equal_range (trid);
    for (auto it = begin; it != end; it++)
```

Reply:
it looks good, I meant to use this when studying C++17, but I forgot. Thanks